x-pack/test/api_integration/apis/apm/agent_configuration.ts

```diff
@@ -210,10 +216,38 @@ export default function agentConfigurationTests({ getService }: FtrProviderConte
       });
 
       it(`should have 'applied_by_agent=true' after supplying etag`, async () => {
+        await searchConfigurations({
+          service: { name: 'myservice', environment: 'development' },
+          etag,
+        });
+
         async function getAppliedByAgent() {
           const { body } = await searchConfigurations({
             service: { name: 'myservice', environment: 'development' },
-            etag,
+          });
+
+          return body._source.applied_by_agent;
+        }
+
+        // wait until `applied_by_agent` has been updated in elasticsearch
+        expect(await waitFor(getAppliedByAgent)).to.be(true);
```

Comment:
do you mind changing this (and other instances) to `hasBeenAppliedByAgent`? Maybe a nit, but I think it will make it more clear how `waitFor` is being used.